GeneralsMD/Code/GameEngineDevice/Source/VideoDevice/FFmpeg/FFmpegVideoPlayer.cpp

```diff
@@ -0,0 +1,550 @@
+//         Private Functions                                               
+//----------------------------------------------------------------------------
+
+
+
+//----------------------------------------------------------------------------
+//         Public Functions                                                
+//----------------------------------------------------------------------------
+
+
+//============================================================================
+// FFmpegVideoPlayer::FFmpegVideoPlayer
+//============================================================================
+
+FFmpegVideoPlayer::FFmpegVideoPlayer()
+{
+
+}
+
+//============================================================================
+// FFmpegVideoPlayer::~FFmpegVideoPlayer
+//============================================================================
+
+FFmpegVideoPlayer::~FFmpegVideoPlayer()
+{
+	deinit();
+}
+
+//============================================================================
+// FFmpegVideoPlayer::init
+//============================================================================
+
+void	FFmpegVideoPlayer::init( void )
+{
+	// Need to load the stuff from the ini file.
+	VideoPlayer::init();
+
+	initializeBinkWithMiles();
+}
+
+//============================================================================
+// FFmpegVideoPlayer::deinit
+//============================================================================
+
+void FFmpegVideoPlayer::deinit( void )
+{
+	TheAudio->releaseHandleForBink();
+	VideoPlayer::deinit();
+}
+
+//============================================================================
+// FFmpegVideoPlayer::reset
+//============================================================================
+
+void	FFmpegVideoPlayer::reset( void )
+{
+	VideoPlayer::reset();
+}
+
+//============================================================================
+// FFmpegVideoPlayer::update
+//============================================================================
+
+void	FFmpegVideoPlayer::update( void )
+{
+	VideoPlayer::update();
+
+}
+
+//============================================================================
+// FFmpegVideoPlayer::loseFocus
+//============================================================================
+
+void	FFmpegVideoPlayer::loseFocus( void )
+{
+	VideoPlayer::loseFocus();
+}
+
+//============================================================================
+// FFmpegVideoPlayer::regainFocus
+//============================================================================
+
+void	FFmpegVideoPlayer::regainFocus( void )
+{
+	VideoPlayer::regainFocus();
+}
+
+//============================================================================
+// FFmpegVideoPlayer::createStream
+//============================================================================
+
+VideoStreamInterface* FFmpegVideoPlayer::createStream( File* file )
+{
+
+	if ( file == NULL )
```

Comment:
Seems to be some inconsistency with the indentation here? Mix of tab and space?

Reply:
Was just going by the github diff where the if block indentation is off compared the to the code below it.